src/vs-bicep/Bicep.VSLanguageServerClient/BicepLanguageServerClient.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.VSLanguageServerClient.ContentType;
+using Bicep.VSLanguageServerClient.MiddleLayerProviders;
+using Bicep.VSLanguageServerClient.ProcessLauncher;
+using Bicep.VSLanguageServerClient.ProcessTracker;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Setup.Configuration;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+using StreamJsonRpc;
+
+namespace Bicep.VSLanguageServerClient
+{
+    [Export(typeof(ILanguageClient))]
+    [ContentType(BicepContentTypeDefinition.ContentType)]
+    public class BicepLanguageServerClient : ILanguageClient, ILanguageClientCustomMessage2
+    {
+        private IClientProcess? _process;
+        private readonly ILanguageClientMiddleLayer _middleLayer;
+        private readonly IProcessTracker _processTracker;
+
+        [ImportingConstructor]
+        public BicepLanguageServerClient(IProcessTracker processTracker)
+        {
+            _processTracker = processTracker;
+
+            var setupConfiguration = new SetupConfiguration();
+            var handleSnippetCompletionsMiddleLayer = new HandleSnippetCompletionsMiddleLayer(setupConfiguration.GetInstanceForCurrentProcess().GetInstallationVersion());
+            var updateFormatSettingsMiddleLayer = new UpdateFormatSettingsMiddleLayer();
+            _middleLayer = new AggregatingMiddleLayer(handleSnippetCompletionsMiddleLayer, updateFormatSettingsMiddleLayer);
+        }
+
+        public string Name => "Bicep Language Server";
+
+        // this is allowed to return null, but can't be marked nullabe due to signature
+        public virtual IEnumerable<string> ConfigurationSections => null!;
```

Comment:
Does it make sense to return an empty enumerable?

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/21d7ef5068fbd7ee78e5fe17a214b167979b6e6a